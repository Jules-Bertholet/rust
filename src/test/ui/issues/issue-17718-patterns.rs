static A1: usize = 1;
static mut A2: usize = 1;
const A3: usize = 1;

fn main() {
    match 1 {
        A1 => {} //~ ERROR: referencing statics in patterns is experimental
        A2 => {} //~ ERROR: mutable statics cannot be referenced in patterns
        A3 => {}
        _ => {}
    }
}
