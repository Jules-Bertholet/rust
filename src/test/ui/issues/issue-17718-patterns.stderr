error[E0658]: referencing statics in patterns is experimental
  --> $DIR/issue-17718-patterns.rs:7:9
   |
LL |         A1 => {}
   |         ^^
   |
   = help: add `#![feature(static_in_pattern)]` to the crate attributes to enable

error[E0158]: mutable statics cannot be referenced in patterns
  --> $DIR/issue-17718-patterns.rs:8:9
   |
LL |         A2 => {}
   |         ^^

error: aborting due to 2 previous errors

Some errors have detailed explanations: E0158, E0658.
For more information about an error, try `rustc --explain E0158`.
